pkg/config/engine/containerd/cgroup_driver.go

```diff
@@ -0,0 +1,53 @@
+/**
+# Copyright 2024 NVIDIA CORPORATION
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+**/
+
+package containerd
+
+import (
+	"os"
+	"path/filepath"
+	"sync"
+)
+
+type cgroupDriver string
+
+const (
+	cgroupDriverCgroupFS = cgroupDriver("cgroupfs")
+	cgroupDriverSystemd  = cgroupDriver("systemd")
+)
+
+var (
+	runningSystemd bool
+	detectSystemd  sync.Once
+)
+
+// getCgroupDriver returns the cgroup driver to use.
+// This is adapted from the containerd implementation.
+// https://github.com/containerd/containerd/blob/1fb1882c7de7239711c343c65fa266a2f1eba9db/internal/cri/server/runtime_config_linux.go#L63-L68
+func getCgroupDriver(root string) cgroupDriver {
```

Comment:
Can we consider limiting the PR to just extracting the options from the defaultRuntime? 

I believe this would suffice when it comes to fixing issues like those we've seen in AKS.  Since the `SystemdCgroup = true` needs to be set explicitly, I think it's safe to assume that either i) the `runc` runtime block would have it set OR ii) the `default_runtime` runtime block , given any containerd config toml

Reply:
The problem is that we don't have access to the default config.toml. If we did, we wouldn't need this logic at all and would just pull it from the config file.